assets/quick-buy.js

```diff
@@ -0,0 +1,67 @@
+if (!customElements.get('quickbuy-modal')) {
+  customElements.define('quickbuy-modal', class QuickBuyModal extends ModalDialog {
+    constructor() {
+      super();
+      this.modalContent = this.querySelector('[id^="QuickBuyInfo-"]');
+    }
+
+    hide(preventFocus = false) {
+      const cartNotification = document.querySelector('cart-notification');
+      if (cartNotification) cartNotification.setActiveElement(this.openedBy);
+      this.modalContent.innerHTML = '';
+      
+      if (preventFocus) this.openedBy = null;
+      super.hide();
+    }
+
+    show(opener) {
+      opener.setAttribute('aria-disabled', true);
+      opener.classList.add('loading');
+      opener.querySelector('.loading-overlay__spinner').classList.remove('hidden');
+
+      fetch(opener.getAttribute('data-product-url'))
+        .then((response) => response.text())
+        .then((responseText) => {
+          const responseHTML = new DOMParser().parseFromString(responseText, 'text/html');
+          const productInfo = responseHTML.querySelector('section[id^="MainProduct-"]');
+
+          this.setInnerHTML(this.modalContent, productInfo.innerHTML);
+          
+          if (window.Shopify && Shopify.PaymentButton) {
+            Shopify.PaymentButton.init();
+          }
```

Comment:
Something to maybe tweak as on the cart page the dynamic checkout button aren't coming up in modal.
![](https://screenshot.click/03-35-1yifn-7st1h.png)

Reply:
I wonder if we can fix that on our side. I don't think we can, but I'll investigate and maybe report it to their team. 👁️👄👁️ 

It looks like the script behaves differently when inside a cart page, and as far as I know, `Shopify.PaymentButton` doesn't monitor new elements on the page by default.